respect-lib-shared/src/commonMain/kotlin/world/respect/shared/viewmodel/manageuser/profile/ProfileViewModel.kt

```diff
@@ -0,0 +1,165 @@
+package world.respect.shared.viewmodel.manageuser.profile
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.toRoute
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.datetime.LocalDate
+import org.jetbrains.compose.resources.getString
+import world.respect.datalayer.oneroster.rostering.model.OneRosterGenderEnum
+import world.respect.shared.generated.resources.*
+import world.respect.shared.navigation.NavCommand
+import world.respect.shared.navigation.ProfileScreen
+import world.respect.shared.navigation.SignupScreen
+import world.respect.shared.navigation.WaitingForApproval
+import world.respect.shared.viewmodel.RespectViewModel
+import world.respect.shared.viewmodel.app.appstate.ActionBarButtonUiState
+
+class ProfileViewModel(
```

Comment:
This should be renamed SignupViewModel - otherwise it is ambiguous - profile implies view your profile.